### PR TITLE
AnimateDiff training improvement

### PR DESCRIPTION
**Pull request by SamitHuang**
# What does this PR do?

Improve animatediff training including parallel training. 
   - mixed precision optimized
   - broken data handle for large scale training
   - parallel training script
   - improved training recipes including random prompt dropout
   - FA supports

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [x] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/animatediff/configs/stable_diffusion/v1-inference-mmv2.yaml [hunk 1]
```diff
@@ -57,7 +57,7 @@ model:
       params:
         embed_dim: 4
         monitor: val/rec_loss
-        use_fp16: False # affect generated image quality on GPUs
+        use_fp16: True # affect generated image quality on GPUs. TODO: make it configurable
```
[the remark this answers is not in the document]
Comment (SamitHuang) [reply examples/animatediff/configs/stable_diffusion/v1-inference-mmv2.yaml:60]: It is configurable in inference. If train with vae fp16, then infer with vae fp16 is fine. This value should be consistent to the training setting.

File: examples/animatediff/README.md [hunk 2]
```diff
@@ -6,9 +6,9 @@ This repository is the MindSpore implementation of [AnimateDiff](https://arxiv.o
 
 - [x] Text-to-video generation with AnimdateDiff v2, supporting 16 frames @512x512 resolution on Ascend 910B, 16 frames @256x256 resolution on GPU 3090
 - [x] MotionLoRA inference
-- [x] AnimateDiff v3
-- [ ] Motion Module Training
-- [ ] Motion LoRA Training
+- [x] Motion Module Training
+- [x] Motion LoRA Training
+- [ ] AnimateDiff v3
```
Comment (wtomin) [inline examples/animatediff/README.md:11]: AnimateDiff v3 is supported now.

**Review by wtomin — APPROVED** [review]

**Review by HaoyangLee — APPROVED** [review]

**Review by Gaohan123 — APPROVED** [review]

File: examples/animatediff/text_to_video.py [hunk 3]
```diff
@@ -298,9 +302,12 @@ def main(args):
         default=None,
         help="if not empty, overwrite the path in configs/prompts/{version}/{task}.yaml",
     )
-    parser.add_argument(
-        "--prompt", type=str, default="", help="Input prompt text. If nt empty, it will overwite the prompt in yaml"
-    )
```
Comment (HaoyangLee) [inline examples/animatediff/text_to_video.py:303]: Seems this deletion is inappropriate.
